featurebyte/query_graph/sql/ast/tile.py

```diff
@@ -46,29 +49,30 @@ class BuildTileNode(TableNode):
 
     @property
     def sql(self) -> Expression:
-        start_date_expr = InternalName.TILE_START_DATE_SQL_PLACEHOLDER
-        start_date_epoch = self.context.adapter.to_epoch_seconds(
-            cast(Expression, parse_one(f"CAST({start_date_expr} AS TIMESTAMP)"))
-        ).sql()
-        timestamp_epoch = self.context.adapter.to_epoch_seconds(
-            quoted_identifier(self.timestamp)
-        ).sql()
-
         input_filtered = self._get_input_filtered_within_date_range()
-        input_tiled = select(
-            "*",
-            f"FLOOR(({timestamp_epoch} - {start_date_epoch}) / {self.frequency}) AS tile_index",
-        ).from_(input_filtered.subquery())
-
-        tile_start_date = f"TO_TIMESTAMP({start_date_epoch} + tile_index * {self.frequency})"
+        tile_index_expr = alias_(
+            expressions.Anonymous(
+                this="F_TIMESTAMP_TO_INDEX",
+                expressions=[
+                    self.context.adapter.convert_to_utc_timestamp(
+                        quoted_identifier(self.timestamp)
+                    ),
+                    make_literal_value(self.time_modulo_frequency),
+                    make_literal_value(self.blind_spot),
+                    make_literal_value(self.frequency_minute),
+                ],
+            ),
+            alias="index",
```

Comment:
Do we need to worry whether `index` may cause name collision here?

Reply:
Yes, that is an existing issue. It should be replaced with an internal column name: https://featurebyte.atlassian.net/browse/DEV-1730